app/src/main/java/com/keylesspalace/tusky/MainActivity.kt

```diff
@@ -294,11 +300,32 @@ class MainActivity : BottomSheetActivity(), ActionButtonActivity, HasAndroidInje
 
                         setupTabs(false)
                     }
-
                     is AnnouncementReadEvent -> {
                         unreadAnnouncementsCount--
                         updateAnnouncementsBadge()
                     }
+                    is NewNotificationsEvent -> {
+                        directMessageTab?.let { tab ->
+                            if (event.accountId == accountManager.activeAccount?.accountId) {
```

Comment:
Don't use `accountManager.activeAccount` here (and elsewhere), as that can change over the life of the activity, and you get the activity thinking it's working with one account and it's actually working with another.

```
private lateinit var activeAccount: AccountEntity

// ...

    // Somewhere in onCreate
    activeAccount = accountManager.activeAccount!!
```

should do it.

Reply:
This is valid

Reply:
Ok, I changed that in MainActivity.

There are two more locations with "activeAccount" though in the NotificationFragment and ConversationFragment on loading.
I left i unchanged there.

When testing this seems to work ok (loading in one account does not delete the badge of the other one).

However I'm not quite sure in both locations when the "activeAccount" is what and if it is ok to directly use accountManager.activeAccount in the fragments.